packages/emotion-primitives/src/emotionPrimitive.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import primitives from 'react-primitives'
+import { renderStylesOnNative } from './emotion-native'
+
+const Platform = primitives.Platform
+
+const TAGS = ['Text', 'View', 'Image']
+
+/*
+  Emotion primitive - Use this function instead of 'styled' to render the styles using emotion on Web, React Native and Sketch.
+
+  Tradeoffs -
+
+  * Passing props in string template literal is not supported yet when used with React Native. But you can pass props when using the object styles.
+  * Compute time is more so the render on Sketch is slow right now.
+  * Need to tweak the webpack config when used with Sketch in order suppress warnings for 'fs' module.
+  * Cannot use shorthand because it requires babel plugin.
+  * Need to pass primitives[tag_name] when withComponent is used.
+
+  Usage -
+
+  const Text = emotionPrimitive('Text')`
+    color: hotpink;
+    font-size: 30px;
+  `
+
+  render(<Text>Hello</Text>)
+
+  const View = emotionPrimitive('View')({
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  }, (props) => {
+    color: props.color
+  })
+
+  render(<View color='mistyrose' />)
+
+  const Title = Text.withComponent('Text')
+
+  render(<Title>Hello</Title>)
+*/
+export const emotionPrimitive = tag => {
+  if (Platform.OS === 'web') {
```

Comment:
imho this can be easily split at build time - we can create 2 separate bundles and point from package.json with main/module and react-native keys appropriately

Reply:
That would be great.